src/game.js

```diff
@@ -1,93 +1,100 @@
+  if (!CHECK_USER_INPUT.test(user_input)) {
+    throw new Error("[ERROR] 잘못된 입력입니다.");
+  } else {
+    for (const INPUT_NUM of user_input) {
+      inputNumber.push(Number(INPUT_NUM));
+    }
+    if (user_input.length !== NUM_DIGITS) {
+      throw new Error("[ERROR] 3자리 수를 입력해주세요.");
+    }
+    if (new Set(inputNumber).size !== NUM_DIGITS) {
+      throw new Error("[ERROR] 중복된 입력입니다.");
+    }
   }
-
-  return inputNumber;
-  // } catch (err) {
-  // Console.print(err.message);
-  // return getUserInput(); // process.exit() 안 쓰고 어떻게 처리해야 하는지 모르겠다.
-  // }
-}
-
-async function startGame() {
-  let computerNumber = generateRandomNumber();
+};
+const playGame = async () => {
+  Console.print("숫자 야구 게임을 시작합니다.");
+  const RANDOM_NUMBER = generateRandomNumber();
   while (true) {
     let inputNumber = await getUserInput();
     let strike = 0;
     let ball = 0;
 
-    for (let i = 0; i < FIT_LENGTH; i++) {
-      if (computerNumber.includes(inputNumber[i])) {
-        if (inputNumber[i] === computerNumber[i]) {
+    for (let i = 0; i < NUM_DIGITS; i++) {
+      if (RANDOM_NUMBER.includes(inputNumber[i])) {
+        if (inputNumber[i] === RANDOM_NUMBER[i]) {
           strike++;
         } else {
           ball++;
         }
       }
     }
-
-    if (strike === 3) {
+    if (strike === NUM_DIGITS) {
       Console.print("3스트라이크!");
       Console.print("3개의 숫자를 모두 맞히셨습니다! 게임 종료");
-
-      let answer = await Console.readLineAsync(
-        "게임을 새로 시작하려면 1, 종료하려면 2를 입력하세요: "
-      );
-      if (answer === "1") {
-        computerNumber = generateRandomNumber();
-        continue;
-      } else if (answer === "2") {
-        break;
-      } else {
-        throw new Error("[ERROR]");
-        // break;
-      }
-    } else if (ball === 0 && strike === 0) {
-      Console.print("낫싱");
-    } else {
-      //if (ball > 0 && strike === 0) {
-      Console.print(
-        (ball > 0 ? `${ball}볼 ` : "") +
-          (strike > 0 ? `${strike}스트라이크` : "")
-      );
-      //} else if (strike > 0 && ball === 0) {
-      //   Console.print(`${strike}스트라이크`);
-      // } else {
-      //   Console.print(`${ball}볼 ${strike}스트라이크`);
-      // }
+      await askForRePlay();
+      break;
     }
+    checkGameScore(strike, ball);
+  }
+};
+
+const checkGameScore = (strike, ball) => {
+  if (ball === 0 && strike === 0) {
+    Console.print("낫싱");
+    return;
+  }
+
+  if (ball > 0 && strike === 0) {
+    Console.print(`${ball}볼`);
+    return;
+  }
+
+  if (strike > 0 && ball === 0) {
+    Console.print(`${strike}스트라이크`);
+    return;
+  }
+
+  if (ball > 0 && strike > 0) {
+    Console.print(`${ball}볼 ${strike}스트라이크`);
+    return;
+  }
+};
+
+const askForRePlay = async () => {
+  const ANSWER = await Console.readLineAsync(
+    "게임을 새로 시작하려면 1, 종료하려면 2를 입력하세요. "
+  );
+  if (ANSWER === "1") {
+    await playGame();
+  } else if (ANSWER === "2") {
+    return;
+  } else {
+    throw new Error("[ERROR]");
   }
```

Comment:
```suggestion
  if (ANSWER === "1") await playGame();
  if (ANSWER === "2") return;
  throw new Error("[ERROR]");
```

이렇게 작성하는 것과 어떤 차이가 있을까요?